### PR TITLE
Fix min_max gpu memory logging bug

### LINKED ISSUE #452: min_max log_gpu_memory option bug
**Describe the bug**
Setting `log_gpu_memory='min_max'` in `Trainer` leads to the following bug.

```
Traceback (most recent call last):
  File "/opt/conda/lib/python3.7/site-packages/pytorch_lightning/trainer/trainer.py", line 347, in fit
    self.single_gpu_train(model)
  File "/opt/conda/lib/python3.7/site-packages/pytorch_lightning/trainer/dp_mixin.py", line 79, in single_gpu_train
    self.run_pretrain_routine(model)
  File "/opt/conda/lib/python3.7/site-packages/pytorch_lightning/trainer/trainer.py", line 467, in run_pretrain_routine
    self.train()
  File "/opt/conda/lib/python3.7/site-packages/pytorch_lightning/trainer/train_loop_mixin.py", line 60, in train
    self.run_training_epoch()
  File "/opt/conda/lib/python3.7/site-packages/pytorch_lightning/trainer/train_loop_mixin.py", line 126, in run_training_epoch
    self.log_metrics(batch_step_metrics, grad_norm_dic)
  File "/opt/conda/lib/python3.7/site-packages/pytorch_lightning/trainer/logging_mixin.py", line 20, in log_metrics
    mem_map = memory.get_memory_profile(self.log_gpu_memory)
  File "/opt/conda/lib/python3.7/site-packages/pytorch_lightning/root_module/memory.py", line 205, in get_memory_profile
    for k, v in memory_map:
ValueError: too many values to unpack (expected 2)
```

**To Reproduce**
On current master, execute the following.
```
    trainer = Trainer(
        ...
        log_gpu_memory='min_max',
        ...
    )
    trainer.fit(model)
```

**Expected behavior**
Log the min/max utilization of gpu memory, as `min_max` option is documented.

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04 
 - Version: Current master

I am working on this issue. Will submit a PR soon.

### PR DESCRIPTION
**Pull request by dreamgonfly**
## Before submitting

- Was this discussed/approved via a Github issue? Yes. #452 
- Did you read the [contributor guideline](https://github.com/williamFalcon/pytorch-lightning/blob/master/.github/CONTRIBUTING.md)? Yes.
- Did you make sure to update the docs?  N/A
- Did you write any new necessary tests? Yes.

## What does this PR do?
Fixes #452 

Additionally, this PR updates & simplifies some code.
1. Use `subprocess.run` as officially recommended by python docs (refer to https://docs.python.org/3/library/subprocess.html#using-the-subprocess-module)
2. Clean up code around gpu memory function.
3. Increase test coverage by adding test for get_memory_profile.

## Did you have fun?
Yes :)

### CODE REVIEW

File: pytorch_lightning/root_module/memory.py [hunk 1]
```diff
@@ -224,17 +215,18 @@ def get_gpu_memory_map():
...
+        encoding='utf-8',
+        capture_output=True,
+        check=True)
     # Convert lines into a dictionary
-    gpu_memory = [int(x) for x in result.strip().split('\n')]
-    gpu_memory_map = {}
-    for k, v in zip(range(len(gpu_memory)), gpu_memory):
-        k = f'gpu_{k}'
-        gpu_memory_map[k] = v
+    gpu_memory = [int(x) for x in result.stdout.strip().split('\n')]
```
Comment (Borda) [inline pytorch_lightning/root_module/memory.py]: instead of `\n` consider using `os.sep`

Comment (dreamgonfly) [reply pytorch_lightning/root_module/memory.py]: Done!

**Comment by dreamgonfly** [comment]
This is metrics logs of training with min_max gpu memory logging option.
It shows two gpus with max & min utilization.

image

File: pytorch_lightning/root_module/memory.py (hunk 1 above)
Comment (dreamgonfly) [reply pytorch_lightning/root_module/memory.py]: Thanks for your feedback. I used `os.linesep` instead.

**Comment by williamFalcon** [comment]
@dreamgonfly love the PR! thanks for the great submission.

@Borda i run every PR through GPU tests btw.

**Comment by Borda** [comment]
> @Borda i run every PR through GPU tests btw.

I know, but for those who do not have avalaible couple of GPUs it is better to ask the author to show it... if you do not mind :)

**Pull request merged**